### PR TITLE
Curating Owners: pkg/api

### PR DESCRIPTION
**Pull request by apelisse**
cc @lavalamp @smarterclayton @erictune @thockin @bgrant0607

In an effort to expand the existing pool of reviewers and establish a
two-tiered review process (first someone lgtms and then someone
experienced in the project approves), we are adding new reviewers to
existing owners files.

If You Care About the Process:
------------------------------

We did this by algorithmically figuring out who’s contributed code to
the project and in what directories.  Unfortunately, that doesn’t work
well: people that have made mechanical code changes (e.g change the
copyright header across all directories) end up as reviewers in lots of
places.

Instead of using pure commit data, we generated an excessively large
list of reviewers and pruned based on all time commit data, recent
commit data and review data (number of PRs commented on).

At this point we have a decent list of reviewers, but it needs one last
pass for fine tuning.

Also, see https://github.com/kubernetes/contrib/issues/1389.

TLDR:
-----

As an owner of a sig/directory and a leader of the project, here’s what
we need from you:

1. Use PR https://github.com/kubernetes/kubernetes/pull/35715 as an example.

2. The pull-request is made editable, please edit the `OWNERS` file to
remove the names of people that shouldn't be reviewing code in the
future in the **reviewers** section. You probably do NOT need to modify
the **approvers** section. Names asre sorted by relevance, using some
secret statistics.

3. Notify me if you want some OWNERS file to be removed.  Being an
approver or reviewer of a parent directory makes you a reviewer/approver
of the subdirectories too, so not all OWNERS files may be necessary.

4. Please use ALIAS if you want to use the same list of people over and
over again (don't hesitate to ask me for help, or use the pull-request
above as an example)

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36525)

### CODE REVIEW

**Comment by apelisse** [comment]
his name is missing from the list, but consider adding "ArtfulCoder"

**Comment by thockin** [comment]
This needs some work.

**Comment by thockin** [comment]
I'm going to approve and fixup later

**Comment by thockin** [comment]
@apelisse actually, can you just keep the top-level OWNERS here, and nix the rest?

File: pkg/api/annotations/OWNERS [hunk 1]
```diff
@@ -0,0 +1,4 @@
+reviewers:
+- dims
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/annotations/OWNERS]: remove, add lavalamp

File: pkg/api/endpoints/OWNERS [hunk 2]
```diff
@@ -0,0 +1,8 @@
+reviewers:
+- thockin
+- smarterclayton
+- mikedanese
+- sttts
+- eparis
+- resouer
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/endpoints/OWNERS]: remove, add lavalamp

File: pkg/api/errors/OWNERS [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- timstclair
+- eparis
+- timothysc
+- dims
+- spxtr
+- hongchaodeng
+- krousey
+- satnam6502
+- cjcullen
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/errors/OWNERS]: remove

File: pkg/api/errors/OWNERS [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+- goltermann
```
Comment (lavalamp) [inline pkg/api/errors/OWNERS]: remove

File: pkg/api/events/OWNERS [hunk 5]
```diff
@@ -0,0 +1,2 @@
+reviewers:
+- kargakis
```
Comment (lavalamp) [inline pkg/api/events/OWNERS]: add lavalamp

File: pkg/api/install/OWNERS [hunk 6]
```diff
@@ -0,0 +1,11 @@
+reviewers:
+- lavalamp
+- smarterclayton
+- deads2k
+- caesarxuchao
+- liggitt
+- nikhiljindal
+- dims
+- krousey
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/install/OWNERS]: remove

File: pkg/api/meta/OWNERS [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+- kargakis
+- janetkuo
+- ncdc
+- eparis
+- dims
+- krousey
+- markturansky
+- fabioy
+- resouer
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/meta/OWNERS]: remove

File: pkg/api/meta/OWNERS [hunk 8]
```diff
@@ -0,0 +1,26 @@
+reviewers:
+- thockin
+- smarterclayton
```
Comment (lavalamp) [inline pkg/api/meta/OWNERS]: add lavalamp

File: pkg/api/pod/OWNERS [hunk 9]
```diff
@@ -0,0 +1,4 @@
+reviewers:
+- smarterclayton
+- kargakis
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/pod/OWNERS]: remove

File: pkg/api/resource/OWNERS [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+- mikedanese
+- saad-ali
+- janetkuo
+- timstclair
+- eparis
+- timothysc
+- jbeda
+- xiang90
+- mbohlool
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/resource/OWNERS]: remove

File: pkg/api/rest/OWNERS [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+- hongchaodeng
+- krousey
+- jszczepkowski
+- euank
+- markturansky
+- fgrzadkowski
+- satnam6502
+- fabioy
+- ingvagabund
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/rest/OWNERS]: remove

File: pkg/api/rest/OWNERS [hunk 12]
```diff
@@ -0,0 +1,35 @@
+reviewers:
+- thockin
+- smarterclayton
```
Comment (lavalamp) [inline pkg/api/rest/OWNERS]: add lavalamp

File: pkg/api/service/OWNERS [hunk 13]
```diff
@@ -0,0 +1,5 @@
+reviewers:
+- justinsb
+- girishkalele
+- freehan
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/service/OWNERS]: remove

File: pkg/api/service/OWNERS [hunk 14]
```diff
@@ -0,0 +1,5 @@
+reviewers:
+- justinsb
+- girishkalele
```
Comment (lavalamp) [inline pkg/api/service/OWNERS]: remove

File: pkg/api/testapi/OWNERS [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+- nikhiljindal
+- bprashanth
+- erictune
+- timstclair
+- eparis
+- soltysh
+- madhusudancs
+- markturansky
+- mml
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/testapi/OWNERS]: remove

File: pkg/api/unversioned/OWNERS [hunk 16]
```diff
@@ -0,0 +1,33 @@
+reviewers:
+- thockin
+- smarterclayton
+- wojtek-t
+- deads2k
+- brendandburns
```
Comment (lavalamp) [inline pkg/api/unversioned/OWNERS]: remove, add lavalamp

File: pkg/api/unversioned/OWNERS [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+- ncdc
+- timothysc
+- soltysh
+- dims
+- madhusudancs
+- hongchaodeng
+- krousey
+- mml
+- mbohlool
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/unversioned/OWNERS]: remove

File: pkg/api/util/OWNERS [hunk 18]
```diff
@@ -0,0 +1,3 @@
+reviewers:
+- caesarxuchao
+- david-mcmahon
```
Comment (lavalamp) [inline pkg/api/util/OWNERS]: remove

File: pkg/api/v1/OWNERS [hunk 19]
```diff
@@ -0,0 +1,41 @@
+reviewers:
+- thockin
+- lavalamp
+- smarterclayton
+- wojtek-t
+- deads2k
+- yujuhong
+- brendandburns
```
Comment (lavalamp) [inline pkg/api/v1/OWNERS]: remove

**Review by lavalamp — CHANGES_REQUESTED** [review]

**Comment by thockin** [comment]
@lavalamp - you can hand edit the PR but it is tedious.  Or we can just use
top-level OWNERS for this dir
